src/Compilers/CSharp/Portable/Syntax/SyntaxFacts.cs

```diff
@@ -543,29 +544,50 @@ internal static bool HasAwaitOperations(SyntaxNode node)
                             });
         }
 
-        internal static bool IsNestedFunction(SyntaxNode child)
+        private static bool IsNestedFunction(SyntaxNode child)
+            => IsNestedFunction(child.Kind());
+
+        private static bool IsNestedFunction(SyntaxKind kind)
+            => kind is SyntaxKind.LocalFunctionStatement
+                or SyntaxKind.AnonymousMethodExpression
+                or SyntaxKind.SimpleLambdaExpression
+                or SyntaxKind.ParenthesizedLambdaExpression;
+
+        [PerformanceSensitive("", Constraint = "Use Green nodes for walking to avoid heavy allocations.")]
```

Comment:
Here's the link to the instructions for collecting similar data.

```suggestion
        [PerformanceSensitive("https://github.com/dotnet/roslyn/pull/66970", Constraint = "Use Green nodes for walking to avoid heavy allocations.")]
```